pulser/waveforms.py

```diff
@@ -409,16 +424,26 @@ class BlackmanWaveform(Waveform):
             case it takes the positive waveform and changes the sign of all its
             values.
     """
-    def __init__(self, duration, area):
+
+    def __init__(self, duration: Union[int, Parametrized],
+                 area: Union[float, Parametrized]):
         """Initializes a Blackman waveform."""
         super().__init__(duration)
-        self._area = float(area)
-        self._norm_samples = np.clip(np.blackman(self._duration), 0, np.inf)
-        self._scaling = self._area / np.sum(self._norm_samples) / 1e-3
+        try:
+            self._area: float = float(cast(float, area))
+        except (TypeError, ValueError):
+            raise TypeError("area needs to be castable to an float but "
```

Comment:
Arg, only a typo left!

```suggestion
            raise TypeError("area needs to be castable to a float but "
```